File: app/src/main/java/org/qp/android/questopiabundle/lib/LibIConfig.java
package org.qp.android.questopiabundle.lib;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

public class LibIConfig implements Parcelable {

    public static final Creator<LibIConfig> CREATOR = new Creator<>() {
        @Override
        public LibIConfig createFromParcel(Parcel in) {
            return new LibIConfig(in);
        }

        @Override
        public LibIConfig[] newArray(int size) {
            return new LibIConfig[size];
        }
    };

    public boolean useHtml;
    public int fontSize;
    public int backColor;
    public int fontColor;
    public int linkColor;

    public LibIConfig() {
    }

    protected LibIConfig(Parcel in) {
        useHtml = in.readByte() != 0;
        fontSize = in.readInt();
        backColor = in.readInt();
        fontColor = in.readInt();
        linkColor = in.readInt();
    }

    public void reset() {
        useHtml = false;
        fontSize = 0;
        backColor = 0;
        fontColor = 0;
        linkColor = 0;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        dest.writeByte((byte) (useHtml ? 1 : 0));
        dest.writeInt(fontSize);
        dest.writeInt(backColor);
        dest.writeInt(fontColor);
        dest.writeInt(linkColor);
    }
}

File: app/src/main/java/org/qp/android/questopiabundle/lib/LibIConfig.kt
package org.qp.android.questopiabundle.lib

import android.os.Parcel
import android.os.Parcelable

data class LibIConfig(
    var useHtml: Boolean = false,
    var fontSize: Long = 0L,
    var backColor: Long = 0L,
    var fontColor: Long = 0L,
    var linkColor: Long = 0L
) : Parcelable {

    constructor(source: Parcel) : this(
        source.readInt() != 0,
        source.readLong(),
        source.readLong(),
        source.readLong(),
        source.readLong()
    )

    fun reset() {
        useHtml = false
        fontSize = 0L
        backColor = 0L
        fontColor = 0L
        linkColor = 0L
    }

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(if (useHtml) 1 else 0)
        dest.writeLong(fontSize)
        dest.writeLong(backColor)
        dest.writeLong(fontColor)
        dest.writeLong(linkColor)
    }

    companion object CREATOR : Parcelable.Creator<LibIConfig> {
        override fun createFromParcel(source: Parcel): LibIConfig {
            return LibIConfig(source)
        }

        override fun newArray(size: Int): Array<LibIConfig?> {
            return arrayOfNulls(size)
        }
    }
}
